### PR TITLE
fix for zigbee2mqtt issues #21791 and #20098

### PR DESCRIPTION
**Pull request by scottyphillips**
Good evening, 

As requested, this is a fix for Mercator Ikuu dimmers SSWM-DIMZ and SSWRM-ZB  (White label Tuya push button or rotary dimmer mechs). I am going to assume the other fingerprinted devices are exhibiting the same issues. 

https://github.com/Koenkk/zigbee2mqtt/issues/21791#issue-2185849747

### CODE REVIEW

File: src/devices/tuya.ts [hunk 1]
```diff
@@ -1410,16 +1410,18 @@ const definitions: Definition[] = [
...
-        configure: async (device, coordinatorEndpoint, logger) => {
-            await tuya.configureMagicPacket(device, coordinatorEndpoint, logger);
-            const endpoint = device.getEndpoint(1);
-            await reporting.bind(endpoint, coordinatorEndpoint, ['genOnOff', 'genLevelCtrl']);
+        fromZigbee: [tuya.fz.datapoints],
+        toZigbee: [tuya.tz.datapoints],
+        configure: tuya.configureMagicPacket,
+        exposes: [tuya.exposes.lightBrightnessWithMinMax()],
+        meta: {
+            tuyaDatapoints: [
```
Comment (Koenkk) [inline src/devices/tuya.ts:1418]: Can you check if the datapoints from `TS0601_dimmer_1` also work?

Comment (scottyphillips) [reply src/devices/tuya.ts:1418]: Hi @Koenkk I updated my external converter with the following from TS0601_dimmer_1:

```
        exposes: [tuya.exposes.lightBrightnessWithMinMax(), tuya.exposes.countdown(), tuya.exposes.lightType(),
            e.power_on_behavior().withAccess(ea.STATE_SET),
            tuya.exposes.backlightModeOffNormalInverted().withAccess(ea.STATE_SET)],
        meta: {
            tuyaDatapoints: [
                [1, 'state', tuya.valueConverter.onOff, {skip: tuya.skip.stateOnAndBrightnessPresent}],
                [2, 'brightness', tuya.valueConverter.scale0_254to0_1000],
                [3, 'min_brightness', tuya.valueConverter.scale0_254to0_1000],
                [4, 'light_type', tuya.valueConverter.lightType],
                [5, 'max_brightness', tuya.valueConverter.scale0_254to0_1000],
                [6, 'countdown', tuya.valueConverter.countdown],
                [14, 'power_on_behavior', tuya.valueConverter.powerOnBehavior],
                [21, 'backlight_mode', tuya.valueConverter.backlightModeOffNormalInverted],
            ],
        },
```
The 'countdown' feature works. All the other options are exposed as configurable and don't throw any UNSUPPORTED ATTRIBUTE errors if I configure them. Backlight I dont think is applicable in my case but it is configurable. I can test Power-on behavior but I need to shut off circuits to do that.

Comment (Koenkk) [reply src/devices/tuya.ts:1418]: Then I would propose to merge both definitions.

**Comment by scottyphillips** [comment]
I have merged 'tuya_dimmer' into 'tuya_dimmer_1' . CI pipeline is failing linting.

**Comment by scottyphillips** [comment]
Good evening, I have merged the two definitions as requested.

**Comment by Koenkk** [comment]
Thanks!

**Comment by JohnnyBreslin** [comment]
Can you confirm if this is likely to resolve the TS0601_dimmer_5 issues withs state:"toggle" and state:"" as well?

Thanks

**Comment by Koenkk** [comment]
@JohnnyBreslin no, tuya devices don't support toggle.

**Comment by JohnnyBreslin** [comment]
Hi Koen,

Thanks for the info, I assume it does not support the status request either so I will do it myself or change the device.

The device, and many others, are described on the Zigbee2mqtt devices as supporting state "toggle" and "". 

Are the manufacturers free to make any claim about their products without any validation or requirement to be accurate?

I will write to them but I assume that as long as they are making sales they won't change the definition and more of us will end up with products that don't operate as described.

Thanks for all of your work on Zigbee2mqtt, it is brilliant and I am a real fan.

Donation on the way, keep up the good work.

**Pull request merged**